python/pyspark/ml/image.py

```diff
@@ -20,6 +20,9 @@
 
     An attribute of this module that contains the instance of :class:`_ImageSchema`.
 
+.. note:: Deprecated in 2.4.0. Use `spark.read.format("image").load(path)` instead and
```

Comment:
FYI, that datasource wouldn't be able to replace `toNDArray` and `toImage` which require numpy. Might be better leave a note that it needs manual conversion with NumPy API for both.